packages/plugins/plugin-advertising-id/src/AdvertisingIdPlugin.ts

```diff
@@ -48,6 +52,16 @@ export class AdvertisingIdPlugin extends Plugin {
       });
   }
 
+  execute(event: SegmentEvent){
+
+    if (this.advertisingId === undefined) {
+      this.queuedEvents.push(event);
+    }else{
+      return event;
+    }
+    return;
```

Comment:
What does it means in this context to return `undefined`?  What if there's other plugin that is meant to be executed after this plugin? Does this means that the other plugin won't get anything to process? Will the other plugin indeed eventually get something once we execute `process` for this event?